mmtrack/datasets/tao_dataset.py

```diff
@@ -0,0 +1,238 @@
+        for idx in range(len(self)):
+            img_id = self.img_ids[idx]
+            result = results[idx]
+            for label in range(len(result)):
+                bboxes = result[label]
+                for i in range(bboxes.shape[0]):
+                    data = dict(
+                        image_id=img_id,
+                        bbox=self.xyxy2xywh(bboxes[i]),
+                        score=float(bboxes[i][4]),
+                        category_id=label
+                        if len(result) == 1230 else self.cat_ids[label])
+                    json_results.append(data)
+        return json_results
+
+    def format_results(self, results, resfile_path=None):
+        """Format the results to json (standard format for COCO evaluation).
+
+        Args:
+            results (list[tuple | numpy.ndarray]): Testing results of the
+                dataset.
+            jsonfile_prefix (str | None): The prefix of json files. It includes
+                the file path and the prefix of filename, e.g., "a/b/prefix".
+                If not specified, a temp file will be created. Default: None.
+
+        Returns:
+            tuple: (result_files, tmp_dir), result_files is a dict containing \
+                the json filepaths, tmp_dir is the temporal directory created \
+                for saving json files when jsonfile_prefix is not specified.
+        """
+        assert isinstance(results, dict), 'results must be a list'
+        assert 'track_bboxes' in results
+        assert 'det_bboxes' in results
+
+        if resfile_path is None:
+            tmp_dir = tempfile.TemporaryDirectory()
+            resfile_path = tmp_dir.name
+        else:
+            tmp_dir = None
+        os.makedirs(resfile_path, exist_ok=True)
+        result_files = dict()
+
+        bbox_results = self._det2json(results['det_bboxes'])
+        result_files['bbox'] = f'{resfile_path}/tao_bbox.json'
+        mmcv.dump(bbox_results, result_files['bbox'])
+
+        track_results = self._track2json(results['track_bboxes'])
+        result_files['track'] = f'{resfile_path}/tao_track.json'
+        mmcv.dump(track_results, result_files['track'])
+
+        return result_files, tmp_dir
+
+    def evaluate(self,
+                 results,
+                 metric=['bbox', 'track'],
+                 logger=None,
+                 resfile_path=None):
+        if isinstance(metric, list):
+            metrics = metric
+        elif isinstance(metric, str):
+            metrics = [metric]
+        else:
+            raise TypeError('metric must be a list or a str.')
+        allowed_metrics = ['bbox', 'track']
+        for metric in metrics:
+            if metric not in allowed_metrics:
+                raise KeyError(f'metric {metric} is not supported.')
+
+        result_files, tmp_dir = self.format_results(results, resfile_path)
+
+        eval_results = dict()
+
+        if 'track' in metrics:
+            from tao.toolkit.tao import TaoEval
```

Comment:
put it in the front of the file, and use `try...except taoeval=None` to import the module